addon/chart-data-updater.js

```diff
@@ -31,7 +58,10 @@ export default Ember.Object.extend({
         if (e instanceof TypeError) { self.set('redraw', true); }
         else { console.error(e); }
       }
+
     });
+	//Return the redraw flag
+    return this.get('redraw');
```

Comment:
Morning Ahmed. I thought the same about this but actually during debugging I was seeing that the calling function in ember-chart.js 
<code>var redraw = ChartDataUpdater['default'].create({
        data: data,
        chart: chart
      }).updateByType(); </code>

was calling but the redraw variable was null. This is because if you look at chart-data-updater.js the lines:-
<code>
updateByType: function updateByType() {

      var data = this.get('data');
      if (data.datasets) {
        return this.updateLinearCharts();
      }
      if (Array.isArray(data)) {
        return this.updatePieCharts();
      }
      return this.get('redraw');
    },
</code>
will never return the redraw variable if either one of the if statements if true. The last line will never get executed because both linear and pie charts if statements return directly from the function call within the if statement. 
( I haven't used the pie chart but a quick peak at the function show me that the same is true there and that updatePieCharts should also have a return statement return the redraw variable).